### PR TITLE
Moves API logic from sponsors route to controller

### LINKED ISSUE #614: Move API logic in sponsors route to controller & stop using route-action
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

**Current behavior:**
The API logic is currently written inside the route.

**Expected behavior:**
The API logic should be inside the controller of the route.

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Moves API logic from sponsors route to controller

Fixes #614

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-615-open-event-frontend-fossasia.surge.sh

**Comment by geekyd** [comment]
@niranjan94 Please review. I have made the changes as suggested in #613

File: app/routes/events/view/edit/sponsors.js [hunk 1]
```diff
@@ -8,48 +8,10 @@ export default Route.extend(EventWizardMixin, {
...
-            }, function() {
-              this.get('notify').error(this.l10n.t('Sponsors data did not save. Please try again'));
-            });
-        })
-        .catch(() => {
-          this.get('notify').error(this.l10n.t('Sponsors data did not save. Please try again'));
-        });
-    }
+    let eventDetails = this.modelFor('events.view.edit');
+    return RSVP.hash({
```
Comment (niranjan94) [inline app/routes/events/view/edit/sponsors.js]: Too many levels of nesting leads to confusion. Instead, 
```js
let data = this.modelFor('events.view.edit');
data.sponsors = data.event.get('sponsors');
return RSVP.hash(data)
```

Comment (geekyd) [inline app/routes/events/view/edit/sponsors.js]: We can go with `data.event.sponsors = data.event.get('sponsors');` instead?

Comment (niranjan94) [inline app/routes/events/view/edit/sponsors.js]: No. Don't touch the event model. Its safer to keep it apart.

**Comment by geekyd** [comment]
@niranjan94 Done!

**Comment by niranjan94** [comment]
Deployment link: https://pr-615-open-event-frontend-fossasia.surge.sh

File: app/components/forms/wizard/sponsors-step.js [hunk 2]
```diff
@@ -23,39 +23,39 @@ export default Component.extend(FormMixin, {
...
   }),
 
   actions: {
     addSponsor() {
-      this.get('data.event.sponsors').addObject(this.store.createRecord('sponsor'));
+      this.get('data.sponsors').addObject(this.store.createRecord('sponsor'));
     },
     removeSponsor(sponsor) {
-      sponsor.deleteRecord();
+      sponsor.destroyRecord();
```
Comment (CosmicCoder96) [inline app/components/forms/wizard/sponsors-step.js]: @geekyd Why this change?

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: Delete does not ensure that the DELETE request is sent as soon as the user clicks the delete button. With `destroyRecord` as soon as the user clicks on delete button, a DELETE request is sent.

PS. `destroyRecord` works better

Comment (CosmicCoder96) [inline app/components/forms/wizard/sponsors-step.js]: @geekyd But isn't the functionality supposed to work like this, even if a user clicks on the delete button of the sponsor, the sponsor should get deleted only if the save  action is triggered

Comment (CosmicCoder96) [inline app/components/forms/wizard/sponsors-step.js]: deleteRecord marks them for deletion and when the save button is pressed, a **DELETE** request is sent.

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @CosmicCoder96 Thats a valid point.
@niranjan94 What do you suggest?

Comment (niranjan94) [inline app/components/forms/wizard/sponsors-step.js]: I agree with @CosmicCoder96 on this ... Let us delete record only when save button is pressed or when users navigates to a different page.

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @niranjan94 Can we do the saving when the user navigates to different page , in a separate issue?
Lets make the event creation work first.

**Comment by niranjan94** [comment]
Deployment link: https://pr-615-open-event-frontend-fossasia.surge.sh

File: app/components/forms/wizard/sponsors-step.js (hunk 2 above)
Comment (CosmicCoder96) [inline app/components/forms/wizard/sponsors-step.js]: @geekyd  I don't understand, the functionality is already there, the change at this line is breaking it please just keep it deleteRecord ?

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @CosmicCoder96 I have already done that. please check

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @CosmicCoder96 I have already done that. please check

Comment (niranjan94) [inline app/components/forms/wizard/sponsors-step.js]: > when users navigates to a different page.
 
By this I mean, a different step.

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @niranjan94 Do we want to send a request to the server on transitions?

Comment (CosmicCoder96) [inline app/components/forms/wizard/sponsors-step.js]: @geekyd oh, sorry. I misunderstood your statement :)

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @niranjan94 @CosmicCoder96 I would suggest to do that as a separate issue & get the event creation working first. What do you guys suggest?

Comment (niranjan94) [inline app/components/forms/wizard/sponsors-step.js]: > Do we want to send a request to the server on transitions?
  
No.

Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: @niranjan94 Okay then it is working as expected please review. Thanks
